modules/hcp-vm-client/main.tf

```diff
@@ -89,6 +89,7 @@ resource "azurerm_linux_virtual_machine" "vm" {
 
   user_data = base64encode(templatefile("${path.module}/templates/user_data.sh", {
     setup = base64gzip(templatefile("${path.module}/templates/setup.sh", {
+      subnet_id        = var.subnet_id,
```

Comment:
Did you end up testing whether this was unique across runs?

Reply:
Good point Chris. As soon as I ran this twice it failed :( because it found multiple instances.